client/softlayer_client_test.go

```diff
@@ -35,6 +35,16 @@ var _ = Describe("SoftLayerClient", func() {
 		})
 	})
 
+	Context("#NewSoftLayerClient_HTTPClient", func() {
```

Comment:
Done. I've added the testing for a custom HTTP client. I've rebased and force pushed, please have check: 

https://github.com/fatih/softlayer-go/commit/d395a37d9d93532cc41c93e864be3101dd5b29fe#diff-7ad39016e99d3c1c7dab0d9b0245f116R50